manager/src/main/java/com/redhat/service/bridge/manager/ProcessorServiceImpl.java

```diff
@@ -112,13 +120,19 @@ public Processor createProcessor(String bridgeId, String customerId, ProcessorRe
         ProcessorDefinition definition = new ProcessorDefinition(requestedFilters, requestedTransformationTemplate, requestedAction, resolvedAction);
         newProcessor.setDefinition(definitionToJsonNode(definition));
 
-        processorDAO.persist(newProcessor);
+        Optional<ConnectorEntity> persist = persistProcessorCreateConnector(newProcessor, actionProvider, resolvedAction);
 
-        connectorService.createConnectorIfNeeded(resolvedAction, newProcessor, actionProvider);
+        persist.ifPresent(c -> eventBus.requestAndForget(CONNECTOR_CREATED_EVENT, c));
```

Comment:
I'm wondering if it's responsability of the `connectorService` to fire the event

Reply:
Doing do will need to call yet another method on that service. A separation of service mapping 1:1 the name with the entities won't scale when we modify one than more entity.